examples/linear_model/plot_lasso_model_selection.py

```diff
@@ -1,171 +1,232 @@
-X /= np.sqrt(np.sum(X ** 2, axis=0))
-
-# #############################################################################
-# LassoLarsIC: least angle regression with BIC/AIC criterion
-
-model_bic = LassoLarsIC(criterion="bic", normalize=False)
-t1 = time.time()
-model_bic.fit(X, y)
-t_bic = time.time() - t1
-alpha_bic_ = model_bic.alpha_
-
-model_aic = LassoLarsIC(criterion="aic", normalize=False)
-model_aic.fit(X, y)
-alpha_aic_ = model_aic.alpha_
-
-
-def plot_ic_criterion(model, name, color):
-    criterion_ = model.criterion_
-    plt.semilogx(
-        model.alphas_ + EPSILON,
-        criterion_,
-        "--",
-        color=color,
-        linewidth=3,
-        label="%s criterion" % name,
-    )
-    plt.axvline(
-        model.alpha_ + EPSILON,
-        color=color,
-        linewidth=3,
-        label="alpha: %s estimate" % name,
-    )
-    plt.xlabel(r"$\alpha$")
-    plt.ylabel("criterion")
-
-
-plt.figure()
-plot_ic_criterion(model_aic, "AIC", "b")
-plot_ic_criterion(model_bic, "BIC", "r")
-plt.legend()
-plt.title("Information-criterion for model selection (training time %.3fs)" % t_bic)
-
-# #############################################################################
-# LassoCV: coordinate descent
+n_random_features = 14
+X_random = pd.DataFrame(
+    rng.randn(X.shape[0], n_random_features),
+    columns=[f"random_{i:02d}" for i in range(n_random_features)],
+)
+X = pd.concat([X, X_random], axis=1)
+X.head()
+
+# %%
+# Selecting Lasso via an information criterion
+# --------------------------------------------
+# :class:`~sklearn.linear_model.LassoLarsIC` provides an Lasso estimator that
+# uses the Akaike information criterion (AIC) or the Bayes information
+# criterion (BIC) to select the optimal value of the regularization
+# parameter alpha.
+#
+# Before fitting the model, we will standardize the data with a
+# :class:`~sklearn.preprocessing.StandardScaler`. In addition, we will
+# measure the time to fit and tune the hyperparameter alpha in order to
+# compare with the cross-validation strategy.
+#
+# We will first fit a Lasso model with the AIC criterion.
+import time
+from sklearn.preprocessing import StandardScaler
+from sklearn.linear_model import LassoLarsIC
+from sklearn.pipeline import make_pipeline
+
+start_time = time.time()
+lasso_lars_ic = make_pipeline(
+    StandardScaler(), LassoLarsIC(criterion="aic", normalize=False)
+).fit(X, y)
+fit_time = time.time() - start_time
+
+# %%
+# We store the AIC metric for each value of alpha used during `fit`.
+results = pd.DataFrame(
+    {
+        "alphas": lasso_lars_ic[-1].alphas_,
+        "AIC criterion": lasso_lars_ic[-1].criterion_,
+    }
+).set_index("alphas")
+alpha_aic = lasso_lars_ic[-1].alpha_
+
+# %%
+# Now, we perform the same analysis using the BIC criterion.
+lasso_lars_ic.set_params(lassolarsic__criterion="bic").fit(X, y)
+results["BIC criterion"] = lasso_lars_ic[-1].criterion_
+alpha_bic = lasso_lars_ic[-1].alpha_
+
+# %%
+results
```

Comment:
Could we highlight the cell with the best AIC and the best BIC value?
Like the table in https://scikit-learn.org/stable/auto_examples/ensemble/plot_gradient_boosting_quantile.html#analysis-of-the-error-metrics.